File: Components/component_base/src/main/java/com/jaydroid/conponent_base/app/delegate/ApplicationDelegate.java
package com.jaydroid.conponent_base.app.delegate;

import android.app.Application;
import android.content.Context;

import java.util.List;

/**
 * Created by COOTEK on 2017/8/29.
 */

public class ApplicationDelegate implements IAppLife {
    private List<IAppLife> list;


    public ApplicationDelegate(Context base) {
        //初始化Manifest文件解析器，用于解析组件在自己的Manifest文件配置的Application
        list = new ManifestParser(base).parse();
    }

    @Override
    public void attachBaseContext(Context base) {
        if (list != null && list.size() > 0) {
            for (IAppLife life : list) {
                life.attachBaseContext(base);
            }
        }
    }

    @Override
    public void onCreate(Application application) {
        if (list != null && list.size() > 0) {
            for (IAppLife life : list) {
                life.onCreate(application);
            }
        }
    }

    @Override
    public void onTerminate(Application application) {
        if (list != null && list.size() > 0) {
            for (IAppLife life : list) {
                life.onTerminate(application);
            }
        }
        list = null;
    }
}

File: Components/component_base/src/main/java/com/jaydroid/conponent_base/app/delegate/ApplicationDelegate.kt
package com.jaydroid.conponent_base.app.delegate

import android.app.Application
import android.content.Context
import android.util.Log

/**
 * Application 代理类
 *
 * @author wangxuejie
 * @version 1.0
 * @date 2019-10-15 10:57
 */
class ApplicationDelegate(base: Context) : IAppLife {


    private var list: List<IAppLife>? = null

    init {
        //初始化Manifest文件解析器，用于解析组件在自己的Manifest文件配置的Application
        list = ManifestParser(base).parse()
    }

    override fun attachBaseContext(base: Context) {
        if (list != null && list!!.isNotEmpty()) {
            for (life in list!!) {
                life.attachBaseContext(base)
            }
        }
    }

    override fun onCreate(application: Application) {
        if (list != null && list!!.isNotEmpty()) {
            for (life in list!!) {
                Log.e(ManifestParser.TAG, "AppLife name: " + life.javaClass.simpleName)
                life.onCreate(application)
            }
        }
    }

    override fun onTerminate(application: Application) {
        if (list != null && list!!.isNotEmpty()) {
            for (life in list!!) {
                life.onTerminate(application)
            }
        }
        list = null
    }
}
